### ISSUE TITLE
tags parameter parsing fix + more robust parsing

### ISSUE THREAD

**Issue by diablodale** [aws/amazon-ecs-cli#952]
#### Context

My goal is to have a key `foo` which has structured data in its value. I prefer JSON. Naturally, JSON will need at least the four non-alphanumeric characters of `,{}:`. EC2 itself supports this; I am successful using the EC2 console UI.

Unfortunately `ecs-cli up --tags` has a parser which prevents `,=` in tag values. This not compliant with AWS tag rules https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions. `=` is explicitly listed as an allowed character by the tagging rules. 

#### Partial Cause

The code twice below contributes to the failure with `=`. 

https://github.com/aws/amazon-ecs-cli/blob/2d98b5ec075190768d9e6ef7ac52b0a5e974c2da/ecs-cli/modules/utils/utils.go#L70

https://github.com/aws/amazon-ecs-cli/blob/2d98b5ec075190768d9e6ef7ac52b0a5e974c2da/ecs-cli/modules/utils/utils.go#L89

A simple fix is to use the following code; split the current `kv` at the first `=` and the value is everything after it until the end of the string

```go
pair := strings.SplitN(kv, "=", 2)
```

#### Alternatives

The above fix for `=` is an easy partial win, I recommend it. Outside of that, alternatives are:

1. Create a new option `--tag` or `--single-tag`. This does _not_ split on `,` and the option can appear multiple times on the command. This and the `=` fix can likely resolve this issue completely. A user can then separately declare the tag/value pairs and escape characters as needed using standard shell methods. E.g.
   ```
   ecs-cli up --single-tag 'keyfoo={"key1":false, "key2":1, "key3":"abcdef"}' \
              --single-tag 'keybar=department_84/33.45'
   ```
2. Create my own structure data format. E.g., use `/` to separate my value entities and `:` separate my key/value pairs. However, now I must maintain my own (de)serialize parser
3. Chose an existing format and base64 encode it. This should work with the existing ecs-cli parser and complies with the above tagging rules. However, the value is not human-readable, can not be easily searched/queried by AWS toolsets, etc.
4. Flatten and do not store structured data in a value. Instead create multiple separate tags each with a single canonical value. This is technically possible, though somewhat undesired because the structured data is highly related to each other.

🤔 I am curious the opinions of others on these alternatives.

#### Has the feature been requested before?

Several closed tagging bugs are in this repo. None exactly this feature.

**Comment by diablodale** [comment]
I have a forthcoming PR to fix the problem with `=` to bring ecs-cli into compliance with AWS tag standards.

```sh
ecs-cli up \
  --tags 'foo-bar=orange=false red-apple=standard,lemon2=yellow'
```

should result in two EC2 tags of

```
tag         value
---------------------
foo-bar     orange=false red-apple=standard
lemon2      yellow
```

**Comment by allisaurus** [comment]
Closing this b/c the fix is now available in [v1.18.1](https://github.com/aws/amazon-ecs-cli/releases/tag/v1.18.1)
